File: src/main/java/recipeapplication/service/IngredientService.java
package recipeapplication.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import recipeapplication.model.User;
import recipeapplication.repository.IngredientRepository;

import javax.transaction.Transactional;

@Service
@Transactional
public class IngredientService
{
    private IngredientRepository ingredientRepository;
    private AuthService authService;

    @Autowired
    public IngredientService(IngredientRepository ingredientRepository, AuthService authService)
    {
        this.ingredientRepository = ingredientRepository;
        this.authService = authService;
    }

    public void deleteAll()
    {
        User user = authService.getLoggedInUser();
        ingredientRepository.deleteAllByUser(user);
    }
}

File: src/main/kotlin/recipeapplication/service/IngredientService.kt
package recipeapplication.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import recipeapplication.repository.IngredientRepository
import javax.transaction.Transactional

@Service
@Transactional
open class IngredientService @Autowired constructor(private val ingredientRepository: IngredientRepository, private val authService: AuthService) {
    fun deleteAll() {
        ingredientRepository.deleteAllByUser(
                authService.loggedInUser
        )
    }
}